### PR TITLE
Add watch script for dev

### PR DESCRIPTION
**Pull request by kaicataldo**
| Q                 | A
| ----------------- | ---
| Bug fix?          | no
| Breaking change?  | no
| New feature?      | no
| Deprecations?     | no
| Spec compliancy?  | no
| Tests added/pass? | no
| Fixed tickets     | comma-separated list of tickets fixed by the PR, if any
| License           | MIT

The watch script was removed in [this PR](https://github.com/babel/babylon/commit/beb8db62646081a88b8db31fe95d4b37aa28d98f) when we started bundling Babylon with rollup (which is awesome!). This PR sets up a watch script for us (the big difference is that with Rollup the `es2015` preset's `modules` option must be `false`, while we need to enable transpiling to CommonJS modules when Babel is run).

 I tried Rollup's watcher, but [it is currently WIP](https://github.com/rollup/rollup-watch), and would crash and was unable to recover from saving files with syntax errors when I tried it.

This is more of a stop gap measure until the Rollup watcher is ready for primetime.

Thoughts on this? I personally was using the watcher and have continued to set up something like this up locally when I'm working on Babylon, so figured it might be nice to have a watch script again. Totally understand we probably want to use one of the two solutions above, but I figured having a watcher is better than not :)

### CODE REVIEW

**Comment by danez** [comment]
It would be nice in general that the tests maybe not use the rollup bundle, so that we can have proper unit tests, right now the only thing that can be tested is the public API.
And then use rollup only for publishing. Although it is probably also nice to know that the rollup bundle works as expected.

**Comment by danez** [comment]
I changed it so that the tests run without rollup.
@DrewML Do you think this is okay?

File: .babelrc [hunk 1]
```diff
@@ -1,16 +1,32 @@
...
+    "production": {
+      "presets": [
+        ["es2015", {
+          "loose": true,
+          "modules": false
+        }],
+        "stage-0"
+      ]
+    },
+    "development": {
```
Comment (kaicataldo) [inline .babelrc]: I wonder if we should just combine `development` and `test, since they're the same? Maybe just `production` and `development?

Comment (kaicataldo) [inline .babelrc]: Oh I see, it's for the `plugins`. Never mind!

**Comment by DrewML** [comment]
> I changed it so that the tests run without rollup.
@DrewML Do you think this is okay?

I think the only part that would be concerning is not running any tests against the bundle in CI. It's possible `rollup` could introduce a bug that we would never catch until running the `Babel` test suite against a release of `Babylon`.

**Comment by kaicataldo** [comment]
> Thinking about it more, we run the Babel test suite in CI against each Babylon PR, right?

Yeah! Seems like we could run that CI test against the bundle?

> It would be nice in general that the tests maybe not use the rollup bundle, so that we can have proper unit tests, right now the only thing that can be tested is the public API.

Could it make sense to split this out into unit and integration/behavioral tests and not bundle for the first and then bundle for the latter?

**Comment by kaicataldo** [comment]
Maybe it would make more sense to separate these two tasks out? Seems like we could just do the watcher script in this one and we then have the env's to take it further for testing, etc. in a future PR?

**Comment by hzoo** [comment]
I think @kaicataldo just wanted to be able to watch in dev as this PR. If it's simple enough we can just land that and then work on the other stuff. Also keep in mind we probably want to just move it back into the monorepo as well 😝

**Comment by danez** [comment]
Yeah true, feel free to reset the branch to your commit and force push.

**Comment by kaicataldo** [comment]
Updated to just add a watch script

File: .babelrc [hunk 2]
```diff
@@ -10,6 +10,14 @@
     "transform-flow-strip-types"
   ],
   "env": {
+    "watch": {
+      "presets": [
+        ["es2015", {
```
Comment (hzoo) [inline .babelrc]: might be able to just do plugins commonjs here due to the inheriting of the top level presets/plugins but this works fine

**Comment by hzoo** [comment]
👍

File: .babelrc (hunk 2 above)
Comment (kaicataldo) [inline .babelrc]: Ohh yeah that could be better :+1:

**Pull request merged**